skrub/_multi_agg_joiner.py

```diff
@@ -0,0 +1,417 @@
+                aux_tables[i] = X
+            elif not hasattr(aux_table, "__dataframe__"):
+                raise ValueError(
+                    "`aux_tables` must be an iterable of dataframes or 'X'."
+                )
+
+        # Polars lazyframes will raise an error here.
+        if not hasattr(X, "__dataframe__"):
+            raise TypeError(f"`X` must be a dataframe, got {type(X)}.")
+        if not all(hasattr(aux_table, "__dataframe__") for aux_table in aux_tables):
+            raise TypeError("`aux_tables` must all be dataframes.")
+
+        if is_pandas(X):
+            if not all(is_pandas(aux_table) for aux_table in aux_tables):
+                raise TypeError("All `aux_tables` must be Pandas dataframes.")
+        if is_polars(X):
+            if not all(is_polars(aux_table) for aux_table in aux_tables):
+                raise TypeError("All `aux_tables` must be Polars dataframes.")
+
+        return X, aux_tables
+
+    def _check_keys(self, main_keys, aux_keys, keys):
+        """Check input keys.
+
+        Parameters
+        ----------
+        main_keys : str, iterable of str, or None
+            Matching columns in the main table. Can be a single column name (str)
+            if matching on a single column: ``"User_ID"`` is the same as
+            ``"[User_ID]"``.
+        aux_keys : iterable of str, iterable of iterable of str, or None
+            Matching columns in the auxiliary table. Can be a single column name (str)
+            if matching on a single column: ``"User_ID"`` is the same as
+            ``"[User_ID]"``.
+        keys : iterable of str, or None
+            Can be provided in place of `main_keys` and `aux_keys` when they are the
+            same. We must provide non-``None`` values for either `keys` or both
+            `main_keys` and `aux_keys`.
+
+        Returns
+        -------
+        main_keys, aux_keys : iterable of str and iterable of iterable of str
+            The correct sets of matching columns to use, each provided as a list of
+            column names.
+        """
+        if keys is not None:
+            if aux_keys is not None or main_keys is not None:
+                raise ValueError(
+                    "Can only pass argument `keys` OR `main_keys` and "
+                    "`aux_keys`, not a combination of both."
+                )
+            main_keys, aux_keys = keys, keys
+        else:
+            if aux_keys is None or main_keys is None:
+                raise ValueError(
+                    "Must pass EITHER `keys`, OR (`main_keys` AND `aux_keys`)."
+                )
+        main_keys = atleast_2d_or_none(main_keys)
+        aux_keys = atleast_2d_or_none(aux_keys)
+        # Check 2d shape
+        if len(main_keys) != len(aux_keys):
+            raise ValueError(
+                "`main_keys` and `aux_keys` have different lengths"
+                f" ({len(main_keys)} and {len(aux_keys)}). Cannot join on different"
```

Comment:
maybe this could be a bit clearer if we check that each key list has the same length as aux_tables and reflect that in the error message

btw I think it's quite unlikely that using key would be possible in most cases for the multiaggjoiner, aux_key + main_key will probably always be used

Reply:
I agree, I now check that they have the same length as `aux_tables`

For the second point, I'll leave it as an option for the user even though it's unlikely -- it's still possible that all tables share the same `key`